extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -115,4 +120,29 @@ declare module 'dataworkspace' {
 		 */
 		element: any;
 	}
+
+	export interface IProjectAction {
+		/**
+		 * id and display name of the project action
+		 */
+		readonly id: string;
+
+		/**
+		 * icon path of the project action
+		 */
+		readonly icon?: string | vscode.Uri | { light: string | vscode.Uri, dark: string | vscode.Uri };
```

Comment:
This is probably a good spot for defining a union type somewhere

`type Icon: string | vscode.Uri | { light: string | vscode.Uri, dark: string | vscode.Uri };`
then later
`readonly icon?: Icon;`

Do we think that might cause too many issues down the line?  Just an idea.

Reply:
If you do this it would probably be best just to add the type to azdata.d.ts since that's where the Icon type is used anyways

Reply:
Created this workitem for this suggestion/idea: https://github.com/microsoft/azuredatastudio/issues/14411